[macOS MiniBrowser] Implement pointer lock client delegates and display banner

aprotyas opened the pull request:
#### 8ce6e671f9d0f1e22d3125f12884efb9996e93bd
<pre>
[macOS MiniBrowser] Implement pointer lock client delegates and display banner
<a href="https://bugs.webkit.org/show_bug.cgi?id=263757">https://bugs.webkit.org/show_bug.cgi?id=263757</a>
rdar://117562146

Reviewed by Aditya Keerthi.

Right now, you can&apos;t enter pointer lock in MiniBrowser because it does
not provide an implementation for the `_webViewDidRequestPointerLock`
client delegate method.

We should address this by providing implementations for both
`_webViewDidRequestPointerLock` and `_webViewDidLosePointerLock`,
allowing us to display a banner that informs the user about their
pointer (and instructs how to exit out of the pointer lock state).

* Source/WebKit/UIProcess/WebPageProxy.cpp:

Drive-by fix to include what condition the `#endif` clause is
terminating.

* Tools/MiniBrowser/mac/WK2BrowserWindowController.m:
(-[WK2BrowserWindowController awakeFromNib]):

Initialize the `_pointerLockBanner` layer here.

(-[WK2BrowserWindowController _webViewDidRequestPointerLock:completionHandler:]):
(-[WK2BrowserWindowController _webViewDidLosePointerLock:]):

Canonical link: <a href="https://commits.webkit.org/269854@main">https://commits.webkit.org/269854@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/29836c1f8dc61f8925098a09437e60e0e95b6403

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/23801 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/1914 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/24898 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/25949 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/21949 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/24076 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/3509 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/24301 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/22476 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/24044 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/1458 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/20576 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/26543 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/1228 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/21487 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/27742 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/21701 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/21765 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/25525 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/1183 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/18877 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/1186 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5692 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/1595 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/1502 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Tools/MiniBrowser/mac/WK2BrowserWindowController.m:
```diff
@@ -50,6 +51,23 @@
 static const int testHeaderBannerHeight = 42;
 static const int testFooterBannerHeight = 58;
 
+static CALayer *makePointerLockBanner(void)
```
Maybe add this as an instance variable in `@implementation WK2BrowserWindowController`? Nice to avoid singletons when possible.

aprotyas replied:
Fixed in [9bcdba2](https://github.com/WebKit/WebKit/pull/19617/commits/9bcdba2b0ce081f28aca988bc21db7e9f711e0d1)

pxlcoder commented on Tools/MiniBrowser/mac/WK2BrowserWindowController.m:
```diff
@@ -143,6 +146,13 @@ - (void)awakeFromNib
+    [_pointerLockBanner setBackgroundColor:NSColor.lightGrayColor.CGColor];
+    [_pointerLockBanner setWrapped:YES];
+    [_pointerLockBanner setFrame:CGRectMake(0, 0, 0, testHeaderBannerHeight)];
+    [_pointerLockBanner setContentsScale:[NSScreen mainScreen].backingScaleFactor];
```
I would have lazily created this in `_webViewDidRequestPointerLock` – but fine if you want to land as-is.

`if (! _pointerLockBanner) ...`

aprotyas replied:
Good idea! Adopted in [29836c1](https://github.com/WebKit/WebKit/pull/19617/commits/29836c1f8dc61f8925098a09437e60e0e95b6403)

aprotyas committed:
[macOS MiniBrowser] Implement pointer lock client delegates and display banner
https://bugs.webkit.org/show_bug.cgi?id=263757
rdar://117562146

Reviewed by Aditya Keerthi.

Right now, you can't enter pointer lock in MiniBrowser because it does
not provide an implementation for the `_webViewDidRequestPointerLock`
client delegate method.

We should address this by providing implementations for both
`_webViewDidRequestPointerLock` and `_webViewDidLosePointerLock`,
allowing us to display a banner that informs the user about their
pointer (and instructs how to exit out of the pointer lock state).

* Source/WebKit/UIProcess/WebPageProxy.cpp:

Drive-by fix to include what condition the `#endif` clause is
terminating.

* Tools/MiniBrowser/mac/WK2BrowserWindowController.m:
(-[WK2BrowserWindowController awakeFromNib]):

Initialize the `_pointerLockBanner` layer here.

(-[WK2BrowserWindowController _webViewDidRequestPointerLock:completionHandler:]):
(-[WK2BrowserWindowController _webViewDidLosePointerLock:]):

Canonical link: https://commits.webkit.org/269854@main
